compose/config/config.py

```diff
@@ -146,7 +146,7 @@ def find(base_dir, filenames):
 
     log.debug("Using configuration files: {}".format(",".join(filenames)))
     return ConfigDetails(
-        os.path.dirname(filenames[0]),
+        os.path.dirname(filenames[0]) if not override_dir else override_dir,
```

Comment:
Similarly, could this be shortened to `override_dir or os.path.dirname(filenames[0])`?